fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/query/plan/cascades/UserDefinedScalarFunction.java

```diff
@@ -100,18 +100,18 @@ public static MacroFunction fromProto(@Nonnull final PlanSerializationContext se
      * Deserializer.
      */
     @AutoService(PlanDeserializer.class)
-    public static class Deserializer implements PlanDeserializer<PMacroFunctionValue, MacroFunction> {
+    public static class Deserializer implements PlanDeserializer<PUserDefinedScalarFunctionValue, UserDefinedScalarFunction> {
```

Comment:
Can we please refrain from using plan serializer logic here. Can you refactor this logic to express the dynamic dispatch in a common way that plan serialization and this logic + @hatyo 's view all use the same underlying logic?